sys/net/network_layer/ng_ipv6/netif/ng_ipv6_netif.c

```diff
@@ -474,14 +475,24 @@ void ng_ipv6_netif_init_by_dev(void)
                            sizeof(if_type)) != -ENOTSUP) &&
             (if_type == NG_NETTYPE_SIXLOWPAN)) {
             uint16_t src_len = 8;
-            DEBUG("Set 6LoWPAN flag\n");
+            uint16_t max_frag_size = UINT16_MAX;
+
+            DEBUG("ipv6 netif: Set 6LoWPAN flag\n");
             ipv6_ifs[i].flags |= NG_IPV6_NETIF_FLAGS_SIXLOWPAN;
             /* use EUI-64 (8-byte address) for IID generation and for sending
              * packets */
             ng_netapi_set(ifs[i], NETCONF_OPT_SRC_LEN, 0, &src_len,
                           sizeof(src_len)); /* don't care for result */
-        }
 
+            if (ng_netapi_get(ifs[i], NETCONF_OPT_MAX_PACKET_SIZE,
+                              0, &max_frag_size, sizeof(max_frag_size)) < 0) {
+                /* if error we assume it works */
```

Comment:
Yes, but why take the minimum, when the maximum is just as good. My point is: I don't see the point where 1280 bytes is a more sensible number than UINT16_MAX or 4242 or 2015 or 5555. The number is quite arbitrary so I decided for the maximum a (standard) IPv6 frame can carry (since its length field is 16-bit long).

Also just to be: this is not the MTU, this is just the number of bytes 6LoWPAN decides to fragment a packet. 6LoWPAN's maximum MTU is `max(max_L2_packet_size, 2^11 - 1)`, because the datagram_length field of the fragmentation header is 11 bit long (but nothing stops it to send bigger packets unfragmented if the link-layer supports that).